### PR TITLE
Add depends_on and in_use_by to product block graphql type

### PR DESCRIPTION
**Pull request by tjeerddie**
Related: #321

### CODE REVIEW

**Review by Mark90 — APPROVED** [review]

File: orchestrator/graphql/schemas/product_block.py [hunk 1]
```diff
@@ -15,4 +15,12 @@ class ProductBlock:
     status: strawberry.auto
     created_at: strawberry.auto
     end_date: strawberry.auto
-    resource_types: Optional[list[ResourceType]]
+    resource_types: list[ResourceType]
+
+    @strawberry.field(description="Return all product blocks that this product block depends on")  # type: ignore
+    async def depends_on(self) -> List[Annotated["ProductBlock", strawberry.lazy(".product_block")]]:
```
Comment (mrijk) [inline orchestrator/graphql/schemas/product_block.py:21]: Mix of list and List?

File: test/unit_tests/graphql/test_product_blocks.py [hunk 2]
```diff
@@ -72,6 +78,103 @@ def test_product_blocks_query(test_client):
...
+            ],
+            "dependsOn": [
+                {"name": "SubBlockOneForTest", "description": "Test Sub Block One"},
+                {"name": "SubBlockTwoForTest", "description": "Test Sub Block Two"},
+            ],
+            "inUseBy": [],
+        },
+    ]
+    for pb in product_blocks:
+        del pb["createdAt"]
```
Comment (mrijk) [inline test/unit_tests/graphql/test_product_blocks.py:174]: This is kinda ugly. Maybe we can have a compare method that has a filter to exclude fields and make this immutable?

**Review by mrijk — APPROVED** [review]

File: orchestrator/graphql/schemas/product_block.py (hunk 1 above)
Comment (tjeerddie) [reply orchestrator/graphql/schemas/product_block.py:21]: thought something would complain about it...

Comment (tjeerddie) [reply orchestrator/graphql/schemas/product_block.py]: changed to `list`

File: test/unit_tests/graphql/test_product_blocks.py (hunk 2 above)
Comment (tjeerddie) [reply test/unit_tests/graphql/test_product_blocks.py]: added in a helper function that uses deepdiff with `exclude_path`

**Pull request merged**